server/datastore/mysql/scheduled_queries.go

```diff
@@ -166,3 +166,15 @@ func (d *Datastore) ScheduledQuery(id uint) (*fleet.ScheduledQuery, error) {
 
 	return sq, nil
 }
+
+func (ds *Datastore) CleanupOrphanScheduledQueryStats() error {
+	_, err := ds.db.Exec(`DELETE FROM scheduled_query_stats where scheduled_query_id not in (select id from scheduled_queries where id=scheduled_query_id)`)
```

Comment:
I like to use `WHERE NOT EXISTS (SELECT 1 FROM table WHERE col=key)` in that scenario, makes the intent a bit clearer IMHO, but that's technically equivalent, feel free to ignore.